src/openloco/localisation/stringmgr.cpp

```diff
@@ -1,23 +1,337 @@
 #include "stringmgr.h"
 #include "../interop/interop.hpp"
+#include "../townmgr.h"
+
+#include <cstring>
+#include <stdexcept>
 
 using namespace openloco::interop;
 
 namespace openloco::stringmgr
 {
+    const uint16_t NUM_USER_STRINGS = 2048;
+    const uint8_t USER_STRING_SIZE = 32;
+
     static loco_global<char * [0xFFFF], 0x005183FC> _strings;
+    static loco_global<char [NUM_USER_STRINGS][USER_STRING_SIZE], 0x0095885C> _userStrings;
 
     const char* get_string(string_id id)
     {
         return _strings[id];
     }
 
+    // TODO: decltype(value)
+    static void format_comma(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x4959A1, regs);
+    }
+
+    // TODO: decltype(value)
+    static void format_int(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495E2A, regs);
+    }
+
+    // TODO: decltype(value)
+    static void formatNumeric_4(uint16_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x4963FC, regs);
+    }
+
+    // TODO: decltype(value)
+    static void format_comma2dp32(uint32_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x4962F1, regs);
+    }
+
+    static void format_comma(uint16_t value, char* buffer)
+    {
+        registers regs;
+        regs.eax = (uint32_t) value;
+        regs.edi = (uint32_t) buffer;
+
+        call(0x495F35, regs);
+    }
+
+    // 0x004958C6
     void format_string(char* buffer, string_id id, void* args)
     {
         registers regs;
         regs.eax = id;
         regs.edi = (uint32_t)buffer;
         regs.ecx = (uint32_t)args;
-        call(0x004958C6, regs);
+
+        if (id >= 0x8000)
+        {
+            if (id < 0x8800)
```

Comment:
Add a note that ID's from `0x8000` to `0x8800` are for user strings. Should probably define the range in appropriate list.